app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -132,22 +131,23 @@ def index_relation_product(query)
     def index_relation_content_view(query)
       if params[:content_view_version_id]
         query = query.where(:content_view_version_id => params[:content_view_version_id])
-        query = Katello::Repository.where(:id => query.select(:library_instance_id)) if params[:library]
+        query = query.archived unless ::Foreman::Cast.to_bool params[:library] || params[:environment_id]
+        query = Katello::Repository.where(:id => query.select(:library_instance_id)) if ::Foreman::Cast.to_bool(params[:library])
```

Comment:
Feel like the meaning of params[:library] changes with this..This would suggest that without if we are not looking for libraries in all environments, we return archived repos. 
Leaning towards a separate archived param to make things less confusing..